src/pybind/mgr/dashboard/services/orchestrator.py

```diff
@@ -1,38 +1,101 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import
 
+from orchestrator import InventoryFilter
 from orchestrator import OrchestratorClientMixin, raise_if_exception, OrchestratorError
 from .. import mgr, logger
+from ..tools import wraps
 
 
 # pylint: disable=abstract-method
-class OrchClient(OrchestratorClientMixin):
+class OrchestratorAPI(OrchestratorClientMixin):
     def __init__(self):
-        super(OrchClient, self).__init__()
+        super(OrchestratorAPI, self).__init__()
         self.set_mgr(mgr)
 
-    def list_service_info(self, service_type):
-        # type: (str) -> list
-        completion = self.describe_service(service_type, None, None)
-        self._orchestrator_wait([completion])
-        raise_if_exception(completion)
-        return completion.result
-
-    def available(self):
+    def status(self):
         try:
-            status, desc = super(OrchClient, self).available()
+            status, desc = super(OrchestratorAPI, self).available()
```

Comment:
In Python you better ask forgiveness than permission, which means: instead of checking whether the Orchestrator is available, you just run your actions, and if not available it will raise an exception.

Reply:
[I agree](http://docs.ceph.com/docs/master/mgr/orchestrator_modules/#orchestrator.Orchestrator.available) but I don't see this used as such here?

Reply:
Yes, you're right. I think I got mixed up here: the `if orch.available():` statements appear twice a few lines below.